test/extensions/filters/http/composite/composite_filter_integration_test.cc

```diff
@@ -291,13 +295,19 @@ class CompositeFilterIntegrationTest : public testing::TestWithParam<CompositeFi
   static std::vector<CompositeFilterTestParams> getValuesForCompositeFilterTest() {
     std::vector<CompositeFilterTestParams> ret;
     for (auto ip_version : TestEnvironment::getIpVersionsForTest()) {
-      CompositeFilterTestParams params;
-      params.version = ip_version;
-      params.is_downstream = true;
-      ret.push_back(params);
-      params.version = ip_version;
-      params.is_downstream = false;
-      ret.push_back(params);
+      for (bool is_downstream : {true, false}) {
+        for (bool is_dual_factory : {true, false}) {
+          if (!is_downstream && !is_dual_factory) {
```

Comment:
This combination(downstream only filter configured in upstream filter chain) is not supported.